app/controllers/doorkeeper_authorizations_controller.rb

```diff
@@ -0,0 +1,22 @@
+class DoorkeeperAuthorizationsController < Doorkeeper::AuthorizationsController
+  before_action :add_ga_to_redirect_uri
+
+protected
+
+  def add_ga_to_redirect_uri
+    if params[:_ga].present? && params[:redirect_uri].present?
+      params[:redirect_uri] =
+        if params[:redirect_uri].include? "?"
+          "#{params[:redirect_uri]}&_ga=#{params[:_ga]}"
+        else
+          "#{params[:redirect_uri]}?_ga=#{params[:_ga]}"
```

Comment:
Isn't there some kind of `CGI.join` function that could help us out here and do the logic for parameter concats for us?

I'll go scout around for some examples when I've hit submit